### PR TITLE
IA-3512 Run welder on Azure VM

### PR DESCRIPTION
**Pull request by Qi77Qi**
https://broadworkbench.atlassian.net/browse/IA-3512

* This sets up welder on Azure VM. The updated welder version should also be compatible with GCP, so it should be safe to merge
* Fix a bug where in [RuntimeV2ServiceInterp.scala](https://github.com/DataBiosphere/leonardo/pull/2804/files#diff-b6c8ddc1824e19ffe2fc9a0ab8b0e372d1cfc4944dc67c23b6f1a5411da904d0), we're using leo's token for WSM calls. But we should use user's access token
* Create a leo staging storage container during VM creation
---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [ ] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/azureModels.scala [hunk 1]
```diff
@@ -3,9 +3,12 @@ package org.broadinstitute.dsde.workbench.leonardo
 import java.util.UUID
 
 final case class WsmControlledResourceId(value: UUID) extends AnyVal
+final case class StorageContainerName(value: String) extends AnyVal
```
Comment (jdcanas) [inline core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/azureModels.scala]: maybe use `ContainerName` from the wb-libs AzureStorage here to cut down on duplicate models

Comment (Qi77Qi) [reply core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/azureModels.scala]: you mean `org.broadinstitute.dsde.workbench.google2.KubernetesSerializableName.ContainerName`? that one doesn't seem right...this is an Azure storage container

File: http/src/main/resources/reference.conf [hunk 2]
```diff
@@ -148,8 +148,15 @@ gce {
...
 azure {
+   wsm {
+     uri = "https://localhost:8000"
+   }
+
    pubsub-handler {
     sam-url = ${sam.server}
+    wsm-url = ${azure.wsm.uri}
+    welder-acr-uri = "terradevacrpublic.azurecr.io/welder-server"
+    welder-image-hash = ${image.welderHash}
```
Comment (jdcanas) [inline http/src/main/resources/reference.conf:159]: hmm.. don't we have this in a specific place already?

I remember our CI that publishes welder hash changes to `reference.conf` had trouble last time we added another welder hash to the config. Could be worth looking into compatibility so we don't have to manually update this too...

Comment (Qi77Qi) [reply http/src/main/resources/reference.conf:159]: yea, it's referecing that exact config you're thinking... which is why value here is ${image.welderHash}

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponent.scala [hunk 3]
```diff
@@ -465,14 +465,29 @@ object clusterQuery extends TableQuery(new ClusterTable(_)) {
...
       // staging bucket is saved as a bucket name rather than a path
-      .map(recs => recs.headOption.flatten.flatMap(head => parseGcsPath("gs://" + head + "/").toOption))
+      .map(recs =>
+        recs.headOption.flatMap { head =>
+          head._1 match {
+            case CloudProvider.Gcp => head._2.map(s => StagingBucket.Gcp(GcsBucketName(s)))
+            case CloudProvider.Azure =>
+              head._2.map { s =>
+                val res = for {
+                  splitted <- Either.catchNonFatal(s.split("/"))
```
Comment (jdcanas) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponent.scala:482]: nit: `splitted` sounds weird (its tense doesn't match the usual words on the left hand side of for comprehensions) can we have it say `split`

Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponent.scala:482]: how about `splittedString` just to be more specific? I think it's fairly common to call it this way

**Review by jdcanas — APPROVED** [review]
Looks good to me pending manual test confirmation....

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponent.scala (hunk 3 above)
Comment (jdcanas) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponent.scala:482]: meh, not super important. just don't like the `ed` which makes it a verb in the past tense, and the other vars are nouns

File: core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/azureModels.scala (hunk 1 above)
Comment (jdcanas) [reply core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/azureModels.scala]: This one https://github.com/broadinstitute/workbench-libs/blob/develop/azure/src/main/scala/org/broadinstitute/dsde/workbench/azure/AzureStorageService.scala#L66

Comment (Qi77Qi) [reply core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/azureModels.scala]: ah cool..I'll pull in the latest wb-libs

**Comment by Qi77Qi** [comment]
jenkins retest

**Comment by Qi77Qi** [comment]
jenkins retest

**Comment by Qi77Qi** [comment]
```
19:14:49 ERROR: for martha-app  Cannot create container for service martha-app: mkdir /var/lib/docker/overlay2/d76d6394b94c4277ff753997825f1e8ae39b0a277ab8747bcda11416162e6393-init: no space left on device
```

jenkins retest

**Pull request merged**